gen.go

```diff
@@ -1686,6 +1686,16 @@ func (t *{{ .Name}}) UnmarshalCBOR(r io.Reader) (err error) {
 		default:
 			return fmt.Errorf("field %q of %q has unsupported kind %q", f.Name, gti.Name, f.Type.Kind())
 		}
+
+		if f.Optional {
+			// If the field implements Initializer by-reference, call Init on the field
+			// to initialize it. Otherwise, we leave it as a zero-value.
+			if _, ok := reflect.Zero(reflect.PointerTo(f.Type)).Interface().(Initializer); ok {
```

Comment:
I tried to do this without creating a zero-value... but I couldn't get it to work with `reflect.Type.Implements`. It looks like _that_ requires the starting type to be an interface in the first place.